src/main.py:71: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  if isinstance(v, collections.Mapping):
[INFO 11:18:54] root Saving to FileStorageObserver in results/sacred.
[DEBUG 11:18:54] pymarl Using capture mode "fd"
[INFO 11:18:54] pymarl Running command 'my_main'
[INFO 11:18:54] pymarl Started run with ID "8"
[DEBUG 11:18:54] pymarl Starting Heartbeat
[DEBUG 11:18:54] my_main Started
[WARNING 11:18:54] my_main CUDA flag use_cuda was switched OFF automatically because no CUDA devices are available!
[INFO 11:18:54] my_main Experiment Parameters:
[INFO 11:18:54] my_main
{   'action_selector': 'soft_policies',
    'add_value_last_step': True,
    'agent': 'rnn',
    'agent_output_type': 'q',
    'batch_size': 10,
    'batch_size_run': 10,
    'buffer_cpu_only': True,
    'buffer_size': 10,
    'checkpoint_path': '',
    'critic_type': 'ac_critic',
    'double_q': True,
    'entropy_coef': 0.01,
    'env': 'griddly',
    'env_args': {   },
    'epochs': 4,
    'eps_clip': 0.2,
    'evaluate': False,
    'gamma': 0.99,
    'grad_norm_clip': 10,
    'hidden_dim': 64,
    'hypergroup': None,
    'hypernet_embed': 64,
    'hypernet_layers': 2,
    'label': 'default_label',
    'learner': 'q_learner',
    'learner_log_interval': 10000,
    'load_step': 0,
    'local_results_path': 'results',
    'log_interval': 50000,
    'lr': 0.0003,
    'mac': 'basic_mac',
    'mask_before_softmax': True,
    'mixing_embed_dim': 32,
    'name': 'qmix',
    'obs_agent_id': True,
    'obs_individual_obs': False,
    'obs_last_action': False,
    'optim_alpha': 0.99,
    'optim_eps': 1e-05,
    'q_nstep': 5,
    'repeat_id': 1,
    'runner': 'episode',
    'runner_log_interval': 10000,
    'save_model': False,
    'save_model_interval': 50000,
    'save_replay': False,
    'seed': 717628456,
    'standardise_returns': False,
    'standardise_rewards': True,
    't_max': 20050000,
    'target_update_interval_or_tau': 0.01,
    'test_greedy': True,
    'test_interval': 50000,
    'test_nepisode': 100,
    'use_cuda': False,
    'use_rnn': False,
    'use_tensorboard': False}
[INFO 11:18:55] my_main Beginning training for 20050000 timesteps
/Users/jonnycook/Desktop/MARLGen/epymarl/src/components/episode_buffer.py:108: UserWarning: Creating a tensor from a list of numpy.ndarrays is extremely slow. Please consider converting the list to a single numpy.ndarray with numpy.array() before converting to a tensor. (Triggered internally at  ../torch/csrc/utils/tensor_new.cpp:201.)
  v = th.tensor(v, dtype=dtype, device=self.device)
[DEBUG 11:18:55] pymarl Stopping Heartbeat
[ERROR 11:18:55] pymarl Failed after 0:00:01!
Traceback (most recent calls WITHOUT Sacred internals):
  File "src/main.py", line 49, in my_main
    run(_run, config, _log)
  File "/Users/jonnycook/Desktop/MARLGen/epymarl/src/run.py", line 56, in run
    run_sequential(args=args, logger=logger)
  File "/Users/jonnycook/Desktop/MARLGen/epymarl/src/run.py", line 183, in run_sequential
    episode_batch = runner.run(test_mode=False)
  File "/Users/jonnycook/Desktop/MARLGen/epymarl/src/runners/episode_runner.py", line 70, in run
    actions = self.mac.select_actions(self.batch, t_ep=self.t, t_env=self.t_env, test_mode=test_mode)
  File "/Users/jonnycook/Desktop/MARLGen/epymarl/src/controllers/basic_controller.py", line 24, in select_actions
    chosen_actions = self.action_selector.select_action(agent_outputs[bs], avail_actions[bs], t_env, test_mode=test_mode)
  File "/Users/jonnycook/Desktop/MARLGen/epymarl/src/components/action_selectors.py", line 73, in select_action
    m = Categorical(agent_inputs)
  File "/Users/jonnycook/opt/anaconda3/envs/MARLGen/lib/python3.7/site-packages/torch/distributions/categorical.py", line 64, in __init__
    super(Categorical, self).__init__(batch_shape, validate_args=validate_args)
  File "/Users/jonnycook/opt/anaconda3/envs/MARLGen/lib/python3.7/site-packages/torch/distributions/distribution.py", line 56, in __init__
    f"Expected parameter {param} "
ValueError: Expected parameter probs (Tensor of shape (1, 2, 5)) of distribution Categorical(probs: torch.Size([1, 2, 5])) to satisfy the constraint Simplex(), but found invalid values:
tensor([[[-0.2088,  1.4356, -0.8478, -0.5731,  1.1941],
         [-0.2997,  1.0617, -0.3718, -0.4765,  1.0863]]],
       grad_fn=<DivBackward0>)
Exception ignored in: <function GymWrapper.__del__ at 0x7fa8534c7e60>
Traceback (most recent call last):
  File "/Users/jonnycook/opt/anaconda3/envs/MARLGen/lib/python3.7/site-packages/griddly/GymWrapper.py", line 413, in __del__
    self.close()
  File "/Users/jonnycook/Desktop/MARLGen/epymarl/src/envs/game_1.py", line 174, in close
    raise NotImplementedError
NotImplementedError: